### PR TITLE
Bug/vspc 186

### PR DESCRIPTION
**Pull request by glend17**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 
https://diging.atlassian.net/browse/VSPC-186
In Users, Users who have staff role was not reflecting appropriately on the dashboard. When clicking Make Staff although the backend would update the role the front-end wouldn't update this.
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

**Comment by jdamerow** [comment]
Please assign two other reviewers and have them review the PR first.

**Review by AvirupBiswas — COMMENTED** [review]
1. vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java

Can you update the comment it is stating "a String of the type of Role we need to validate" but according to code it is expecting a string of value 'ROLE_STAFF'

roleAsString is not used anywhere in UsersUtility.java


2. vspace/src/test/java/edu/asu/diging/simpleusers/web/admin/UsersUtilityTest.java

Method name test_checkUserRoleExists_roleExists needs proper camel casing like that of the other method.

**Comment by glend17** [comment]
@AvirupBiswas 
> 1. vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java
> 
> Can you update the comment it is stating "a String of the type of Role we need to validate" but according to code it is expecting a string of value 'ROLE_STAFF'
> Since this is a generic utility method that can be extended to any role I have left it as that, for the current use case it is for ROLE_STAFF, but can be extended to be used for other roles as well. What do you feel?
> roleAsString is not used anywhere in UsersUtility.java
> Thanks, I have removed this
> 1. vspace/src/test/java/edu/asu/diging/simpleusers/web/admin/UsersUtilityTest.java
> 
> Method name test_checkUserRoleExists_roleExists needs proper camel casing like that of the other method.
> Thanks, have addressed this

**Comment by AvirupBiswas** [comment]
> @AvirupBiswas
> 
> > 1. vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java
> > 
> > Can you update the comment it is stating "a String of the type of Role we need to validate" but according to code it is expecting a string of value 'ROLE_STAFF'
> > Since this is a generic utility method that can be extended to any role I have left it as that, for the current use case it is for ROLE_STAFF, but can be extended to be used for other roles as well. What do you feel?
This is fine.
> > roleAsString is not used anywhere in UsersUtility.java
> > Thanks, I have removed this
> > 
> > 1. vspace/src/test/java/edu/asu/diging/simpleusers/web/admin/UsersUtilityTest.java
> > 
> > Method name test_checkUserRoleExists_roleExists needs proper camel casing like that of the other method.
> > Thanks, have addressed this

**Review by AvirupBiswas — APPROVED** [review]

File: vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+/**
+ * Utility class that contains methods for ListUsersController
+ */
+public class UsersUtility {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java:12]: javadoc needs author information

File: vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    * a String of the type of Role we need to validate.
+    * @param  userRoles  a set of SimpleGrantedAuthority containing details of the roles assigned to a User
+    * @param  role the role that needs to be checked against userRoles
+    * @return      Boolean: True if User is assigned the role, else False.
+    */
+    public static Boolean checkUserRoleExists(Set<SimpleGrantedAuthority> userRoles, String role) {
+        for (SimpleGrantedAuthority roleKey : userRoles) {
+            if(roleKey.toString().equals(role)) return true;  
+        }
+        return false;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java:27]: this can be a one liner using the stream api

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java (hunk 2 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java:26]: Have addressed this now

File: vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java (hunk 1 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java:13]: Have addressed this now

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java (hunk 2 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/simpleusers/web/admin/UsersUtility.java]: My bad, had missed the appropriate return statement, fixed it now.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**